src/services/survey.ts

```diff
@@ -23,16 +25,83 @@ export const listSurveys = async (options = {}): Promise<SurveyResponse> => {
     ...options,
   });
 
+  const deserializedResponse = await deserialize(response);
+
   return {
-    surveys: response.data.map((survey: any) => parseSurvey(survey)),
+    surveys: deserializedResponse.data.map((survey: any) =>
+      parseSurvey(survey)
+    ),
     batchInfo: parseBatchInfo(response.meta),
   };
 };
 
+export const getSurveyDetail = async (surveyId: string) => {
+  const { accessToken } = getUserToken();
+
+  const response = await get(`/surveys/${surveyId}`, {
+    headers: { Authorization: `Bearer ${accessToken}` },
+  });
+
+  const deserializedResponse = await deserialize(response);
+
+  return parseSurveyDetail(deserializedResponse.data);
+};
+
 const parseSurvey = (surveyResponse: any): Survey => ({
   id: surveyResponse.id,
-  title: surveyResponse.attributes.title,
-  description: surveyResponse.attributes.description,
-  coverImageThumbnailUrl: surveyResponse.attributes.coverImageUrl,
-  coverImageUrl: `${surveyResponse.attributes.coverImageUrl}l`,
+  title: surveyResponse.title,
+  description: surveyResponse.description,
+  coverImageThumbnailUrl: surveyResponse.coverImageUrl,
+  coverImageUrl: `${surveyResponse.coverImageUrl}l`,
+});
+
+const parseSurveyDetail = (surveyDetailResponse: any): SurveyDetail => {
+  let surveyIntro, surveyOutro;
+
+  let questions = surveyDetailResponse.questions.map((question: any) =>
+    parseSurveyQuestion(question)
+  );
+
+  [[surveyIntro], questions] = extractQuestionType(questions, 'intro');
+  [[surveyOutro], questions] = extractQuestionType(questions, 'outro');
```

Comment:
I should first say thanks because I finally took the time to play and learn lodash thanks to this PR :D 

Now, I'm wondering if using `partition` as a filter is not a little bit overkill... 
What about using filters only?

```suggestion
  surveyIntro = questions.filter(q => q.type === 'intro');
  surveyOutro = questions.filter(q => q.type === 'outro');
  questions = questions.filter(q => q.type !== 'intro' && q.type !== 'outro');
```

Reply:
@malparty At first, I consider the filter function, but I don't want to iterate the whole question list when extracting the intro and outro. Thinking again, I think it's fine because the question list is not large.

Reply:
@hanam1ni  And maybe also because the "partition" helper needs to iterate the whole questions array as well ;-)
By the way, it is very interesting to dive in `lodash` source code: [The partition helper](https://github.com/lodash/lodash/blob/master/partition.js) is using [a custom "reduce"](https://github.com/lodash/lodash/blob/2da024c3b4f9947a48517639de7560457cd4ec6c/.internal/arrayReduce.js#L19) method that finally does a `while` covering all indexes of the array. 😇 

<img width="480" alt="image" src="https://user-images.githubusercontent.com/77609814/175907530-8c9cf241-e11b-4651-91ec-15cf1b700b86.png">